src/evict/evict_lru.c

```diff
@@ -2211,16 +2213,18 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
      * If we couldn't find the number of pages we were looking for, skip the tree next time.
      */
     if (pages_queued < target_pages / 2 && !urgent_queued)
-        btree->evict_walk_period = WT_MIN(WT_MAX(1, 2 * btree->evict_walk_period), 100);
+        __wt_atomic_store32(&btree->evict_walk_period,
+          WT_MIN(WT_MAX(1, 2 * __wt_atomic_load32(&btree->evict_walk_period)), 100));
     else if (pages_queued == target_pages) {
-        btree->evict_walk_period = 0;
+        __wt_atomic_store32(&btree->evict_walk_period, 0);
         /*
          * If there's a chance the Btree was fully evicted, update the evicted flag in the handle.
          */
         if (__wt_btree_bytes_evictable(session) == 0)
             FLD_SET(session->dhandle->advisory_flags, WT_DHANDLE_ADVISORY_EVICTED);
-    } else if (btree->evict_walk_period > 0)
-        btree->evict_walk_period /= 2;
+    } else if (__wt_atomic_load32(&btree->evict_walk_period) > 0)
```

Comment:
Another case for a local variable.

Reply:
Done in [c088090](https://github.com/wiredtiger/wiredtiger/pull/10635/commits/c08809048d3122fa26da500e7ecda927e373e41d)